### PR TITLE
Python docs

### PR DESCRIPTION
**Pull request by ANekhai**
My draft of documentation for using Python on the HPC. The bulk of the changes live on their own page. I've updated the onDemand docs to include using VSCode and Python and fixed a link in the available software page to point to venv documentation.

Any suggestions for changes and modifications are welcome! I've included sections for FAQs and AI/ML stuff on the python page. I'm happy to expand on these if necessary, or to merge them into other sections if things are too busy!

### CODE REVIEW

**Review by aebruno — APPROVED** [review]

**Review by dsajdak — CHANGES_REQUESTED** [review]
@ANekhai as we discussed here's my full review.  Please verify that the links I added actually work.  :)  I tend to get the formatting wrong at least once!

File: pages/portals/ood.md [hunk 1]
```diff
@@ -126,6 +126,9 @@ When the Jupyter app starts, it will launch these additional modules with it and
 **Loading additional software modules with VSCode:**  
 This works the same way as described above for the Jupyter Lab/Notebook app.  Any software modules listed in this box will be loaded when the VSCode session starts.  
 
+There are a few oddities to note about how VSCode functions at CCR:
```
Comment (dsajdak) [inline pages/portals/ood.md:129]: This section is still not showing up as a list - they're all in a single paragraph

**Review by dsajdak — CHANGES_REQUESTED** [review]
@ANekhai this looks fantastic!  There's just one small issue remaining before we publish.

**Review by dsajdak — APPROVED** [review]
LGTM!

**Pull request merged**